text_collector_examples/storcli.py

```diff
@@ -57,6 +59,13 @@ def main(args):
 
     print_all_metrics(metric_list)
 
+def handle_common_controller(response):
+    (controller_index, baselabel) = get_basic_controller_info(response)
+
+    # Split up string to not trigger CodeSpell issues
+    if 'ROC temperature(Degree Celc' + 'ius)' in response['HwCfg'].keys():
```

Comment:
@yosefy This commit breaks things for me on `Storage Command Line Tool  Ver 1.21.06 Nov 18, 2016` with LSI MegaRAID SAS 9260-4i. Key errors in the if condition will only be captured by the `except` in `main()`. As a result, I only get the basic controller info as output from the script.

Reply:
@Tolsto  but it was same before actually. what do you mean?
there is no try also inside handle_megaraid_controller()
so i don't see where we get the error.
may be you may provide the output of the cli tool?
anyhow what do you suggest to do?

Reply:
The code in line 66 or 67 throws a KeyError for me which gets only rescued in line 57, i.e. no metrics but the controller info metric in output. A possible solution would be to add a try within `handle_common_controller` so that `handle_megaraid_controller` will actually be reached.

This is my entire output with your version:

```
# HELP megaraid_controller_info MegaRAID controller info
# TYPE megaraid_controller_info gauge
megaraid_controller_info{controller="0",model="LSI MegaRAID SAS 9260-4i",serial="XXXXXXX",fwversion="2.130.403-4660"} 1.0
```

Reply:
ah, it means that you don't have response['HwCfg'] key at all
ok lets put it in try block.
@SuperQ , is it ok by you?

Reply:
Yup, sounds fine.

Reply:
ok i have created another PR https://github.com/prometheus/node_exporter/pull/1410/commits/905ae291ea4446a6a2663c1e6728d9d85df9f15e